src/callbacks.rs

```diff
@@ -163,36 +177,87 @@ pub extern fn view_request_geometry(_view: WlcView, _geometry: &Geometry) {
 }
 
 pub extern fn pointer_button(view: WlcView, _time: u32,
-                         mods: &KeyboardModifiers, _button: u32,
+                         mods: &KeyboardModifiers, button: u32,
                              state: ButtonState, point: &Point) -> bool {
-    if state == ButtonState::Pressed && !view.is_root() {
-        if let Ok(mut tree) = try_lock_tree() {
-            tree.set_active_view(view)
-                .unwrap_or_else(|err| {
-                    match err {
-                        TreeError::ViewNotFound(handle) => {
-                            if handle.get_class() != "Background" {
-                                error!("Could not set active container {:?}", err);
-                            }
-                        },
-                        err => error!("Could not set active container {:?}", err)
-                    }
-                });
+    if state == ButtonState::Pressed {
+        if button == 0x110 && !view.is_root() {
+            if let Ok(mut tree) = try_lock_tree() {
+                tree.set_active_view(view)
+                    .unwrap_or_else(|err| {
+                        error!("Could not set active container {:?}", err);
+                    });
+                if mods.mods.contains(MOD_CTRL) {
+                    let action = Action {
+                        view: view,
+                        grab: *point,
+                        edges: ResizeEdge::empty()
+                    };
+                    set_performing_action(Some(action));
+                }
+            }
+        } else if button == 0x111 && !view.is_root() {
+            if let Ok(mut tree) = try_lock_tree() {
+                tree.set_active_view(view)
+                    .unwrap_or_else(|err| {
+                        error!("Could not set active container {:?}", err);
```

Comment:
There's could be a lot of things that could go wrong with setting an active view (eg, there might be an issue setting the active path. Right now the active path isn't allowed to go into floating containers, and this could explain WHY the user facing issue (eg the inability to focus on something) happened)